<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title>Taking a network view of EHR and Biobank data to find explainable multivariate patterns</title>
    <meta charset="utf-8" />
    <meta name="author" content="Nick Strayer" />
    <meta name="date" content="2019-04-03" />
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/default-fonts.css" rel="stylesheet" />
    <link href="libs/font-awesome-5.3.1/css/fontawesome-all.min.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Taking a network view of EHR and Biobank data to find explainable multivariate patterns
### Nick Strayer
### 2019-04-03

---





---
# Talk outline

![:space 3]()
- Electronic medical records

- PheWAS

- Changing how we _think about_ these data

- Multimorbidity explorer

- Changing how we _model_ these data

- The stochastic block model

- Applying on real data

- Future directions



---

# Electronic medical records (EHR)


- In an effort to make healthcare more efficient EHR systems have become common in the US. 

- While originally made for billing purposes there is still a huge sum of information that, with careful effort, _hopefully_, can be extracted for research.

- In this presentation I will focus on the subset of EHR pertaining to billing codes: ICD9, ICD10, and Phecodes.

![:centerPic 30](https://securecdn.pymnts.com/wp-content/uploads/2018/03/bigdata.jpg) ![:small 0.6]([image source](https://www.pymnts.com/data/2018/datatorrent-jeff-bettencourt-real-time-big-data/))



---
# Biobanks

![:space 4] 

<i class="fas  fa-piggy-bank fa-lg fa-fw " style="color:#e34a33;"></i> Some hospitals have repositories of biological samples that can be matched to their EHR.

![:space 8] 

<i class="fas  fa-syringe fa-lg fa-fw " style="color:#e34a33;"></i> Data could be anything from plain unprocessed-plasma all the way to full single-cell sequencing. 

![:space 8] 

<i class="fas  fa-dna fa-lg fa-fw " style="color:#e34a33;"></i> Here I will focus on plain SNP-chip readings, aka presence or absence of a given marker at multiple points on the genome.


---


# PheWAS

In an effort to extract information from these results the technique PheWAS was made.

![:space 8] 

![:borderedCenterPic 80](figures/phewas-paper-head.png)

---

## Concept

![:centerPic 90](figures/phewas-explainer.svg)

---

## The univariate problem

![:space 12]()

<i class="fas  fa-hand-point-up fa-lg fa-fw " style="color:#e34a33;"></i> PheWAS looks at one genome <i class="fas  fa-arrow-right "></i> phenome association at a time.



![:space 12]()

.pull-left[
<i class="fas  fa-stream fa-lg fa-fw " style="color:#e34a33;"></i> This gives us the multiple-comparisons problem.
]
.pull-right[
<i class="fas  fa-project-diagram fa-lg fa-fw " style="color:#e34a33;"></i>Also, does the world work like this? 
]


---


# Changing how we  ![:colorText orangered](think about) these data

![:centerPic 95](figures/obs-2-network.svg)

---
# Multimorbidity explorer

![:space 14]()

![:borderedCenterPic 80](figures/me_paper_abstract.png)

---
## What it is

![:space 5]()

Application that allows researchers to explore the results of PheWAS studies along with investigating individual-level data that produced those results.
    
    
![:space 5]()

.pull-left[
#### <i class="fas  fa-hand-point-up fa-lg fa-fw " style="color:#e34a33;"></i> Interact with results 

PheWAS results are typically delivered with static plots and tables. ME allows researchers to interact with those results.

]

.pull-right[
#### <i class="fas  fa-expand-arrows-alt fa-lg fa-fw " style="color:#e34a33;"></i> Expand past plain associations

By giving researcher's the ability to look at the network behavior of genotype-phenotype associations, it can provide more nuanced insights from the data than a table. 
]


---
## How it is

![:space 5]()


<i class="fas  fa-box-open fa-lg fa-fw " style="color:#e34a33;"></i> Central package contains all the building blocks of a typical ME deployment.

![:space 3]()

<i class="fas  fa-puzzle-piece fa-lg fa-fw " style="color:#e34a33;"></i> Creating a custom app is an exercise in combining the neccesary components. 

![:space 3]()

<i class="fas  fa-cloud fa-lg fa-fw " style="color:#e34a33;"></i> Hosted on lab RStudio Connect instance running on AWS.
  
![:indent]()  <i class="fab  fa-docker fa-lg fa-fw " style="color:#e34a33;"></i> If more security is desired, Docker image is available.

---
## Demo

---


# Changing how we ![:colorText orangered](model) these data
![:space 3]()

<i class="fas  fa-project-diagram fa-lg fa-fw " style="color:#e34a33;"></i> If visualizing the data as a network can help us understand them, why not model them as a network, too?
![:space 4]()

<i class="fas  fa-level-up-alt fa-lg fa-fw " style="color:#e34a33;"></i> Mathematical and statistical models of networks are a diverse and booming field. 
![:space 4]()

<i class="fas  fa-frown-open fa-lg fa-fw " style="color:#e34a33;"></i> Unfortunately, many of the methods sit on fragile methodological foundations.

![:indent]() <i class="fas  fa-grin fa-lg fa-fw " style="color:#e34a33;"></i> Luckily, a new model type has taken the stage recently...

---


## The ![:colorText orangered](S)tochastic ![:colorText orangered](B)lock ![:colorText orangered](M)odel

![:centerPic 90](figures/the-sbm.svg)

---
## SBM formula

![:centerPic 100](figures/sbm-formula.svg)

---
## Bipartite expansion

In its basic form the SBM only works with a single type of node, but we have two (phenotype and patients)...


![:space 4]()


![:borderedCenterPic 80](figures/bipartite-sbm-paper.png)

![:space 4]()

<i class="fas  fa-wrench fa-lg fa-fw " style="color:#e34a33;"></i> To fix this all we need to do is add a constraint to the model to only cluster nodes of the same type.

![:indent]()  &lt;img src='figures/bell-curve.svg' width = 40px/&gt; This is equivalent to an infinitely strong prior on cluster 'types'.


---
## Bipartite SBM formula

![:centerPic 100](figures/bipartite-sbm-formula.svg)



---
## Statistical properties

Model is fit in a bayesian manor

.pull-left[

### Priors
Priors are set on both the group/cluster structure and the edge counts between groups.
]

.pull-right[

### Fitting
Metropolis-Hastings is used to fit model using MCMC

Very similar to how a Dirichlete mixture model is fit

]

---
### Prior on clusters

![:centerPic 90](figures/group-prior.svg)


---

### Prior on edge counts


![:centerPic 90](figures/edge-prior.svg)

---
## Simulation results

![:space 2]()

![:centerPic 65](figures/biSBM_simulations.png)

The BiSBM outperforms dirichlet mixture models even when the true generating distribution is dirichlet. 

![:space 3]()
.right[
![:small 0.9](Figure from A network approach to topic models, Gerlach et al 2019)
]

---

# Applying in the real world



---

# Future directions
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script src="macros.js"></script>
<script>var slideshow = remark.create({
"css": ["styles.css", "metropolis"],
"highlightStyle": "github",
"highlightLines": true
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
